src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -1708,14 +1708,18 @@ public ClusteringComparator getComparator()
 
     public void snapshotWithoutFlush(String snapshotName)
     {
-        snapshotWithoutFlush(snapshotName, null, false);
+        snapshotWithoutFlush(snapshotName, null, false, null);
     }
 
     /**
      * @param ephemeral If this flag is set to true, the snapshot will be cleaned during next startup
      */
-    public Set<SSTableReader> snapshotWithoutFlush(String snapshotName, Predicate<SSTableReader> predicate, boolean ephemeral)
+    public Set<SSTableReader> snapshotWithoutFlush(String snapshotName, Predicate<SSTableReader> predicate, boolean ephemeral, RateLimiter rateLimiter)
     {
+        if (rateLimiter == null)
+            rateLimiter = RateLimiter.create(DatabaseDescriptor.getSnapshotLinksPerSecond() == 0 ?
```

Comment:
Pull this logic out, next to `DatabaseDescriptor#{get,set}SnapshotLinksPerSecond`? So that the special meaning of `0` is apparent and the logic isn't copied three extra times?